ccp/lib/clients/jenkins/base.py

```diff
@@ -0,0 +1,62 @@
+from ccp.lib.clients.base import APIClient
+from ccp.lib.clients.openshift.client import OpenShiftCmdClient
+from ccp.lib.utils.authorization import Authorization, BearerAuthorization
+
+
+def jenkins_jobs_from_jobs_ordered_list(nested_job_ordered_list):
+    """
+    Formats query part of URL from ordered job list as /job/j1/job/j2
+    :param nested_job_ordered_list: The ordered list of jenkins job names,
+    with parent as first and every child below.
+    :type nested_job_ordered_list list
+    :return: A string of the form /job/j1/job/j2...
+    """
+    dest = ""
+    for i in nested_job_ordered_list:
+        dest = "{}/job/{}".format(
+            dest,
+            i
+        )
+    return dest
+
+
+class OpenshiftJenkinsBaseAPIClient(APIClient):
+
+    def __init__(
+            self,
+            server,
+            secure=True,
+            verify_ssl=True,
+            token_from_mount=None,
+            sa="sa/jenkins",
+            namespace="default"
```

Comment:
umm - we should make this is as required parameter IMO.

Reply:
as in, without any defaults

Reply:
We'are already discussing about spreading the jobs in equal numbers across different namespaces, then there would be need to invoke the constructor with different namespaces, the default will rarely be used I think, its better to have this var definition without a default value.

Reply:
Fixing